src/tracks/playabletrack/wavetrack/ui/SpectrumView.h

```diff
@@ -11,32 +11,120 @@ Paul Licameli split from WaveTrackView.h
 #ifndef __AUDACITY_SPECTRUM_VIEW__
 #define __AUDACITY_SPECTRUM_VIEW__
 
+#include <map>
+#include <set>
 #include "WaveTrackView.h" // to inherit
 
+
 class WaveTrack;
+class BrushHandle;
+using TimeFreqBinsMap = std::map<long long, std::set<wxInt64>>;
+
+class SpectralData{
+private:
+   double mSampleRate;
+   long long mStartT = std::numeric_limits<long long>::max();
+   long long mEndT = 0;
```

Comment:
So start is minimum of all sample times so far entered, end is max?  Comment that.